Add hotspot analysis using perf

m-kahla opened the pull request:
Added functionality to benchmark the builds using perf and display in the reports

ggeorgakoudis reviewed (changes requested):
Diff shows a lot of changes that are only about formatting. Can you please revert those changes since they are unrelated to the functionality added?

Commit:
Use 'env' as a separate key that has the env variables in the input YAML file
Add new option to run experiments with perf and get the hotness of each line of code
Add Hotness Columns for each build in HTML reports of Source Code
Only show Hotness coulmn in the reports if we run with perf
